syntax = "proto3";

package ffsrv;

service MyService
{
    rpc Login       (LoginRequest)       returns (LoginResponse)       {}
    rpc Logout      (LogoutRequest)      returns (LogoutResponse)      {}
    rpc AddLocation (AddLocationRequest) returns (AddLocationResponse) {}
}

message LoginRequest
{
    string caller_identity = 1;
}

message LoginResponse
{
    uint64 session_handle = 1;
}

message LogoutRequest
{
    uint64 session_handle = 1;
}

message LogoutResponse
{
    bool succeeded = 1;
}

message Location
{
    double x = 1;
    double y = 2;
    double z = 3;
    string user_data = 4;
}

message AddLocationRequest
{
    uint64 session_handle = 1;
    Location location = 2;
}

message AddLocationResponse
{
    bool succeeded = 1;
}
